tests/lint/genDiff.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: UTF-8 -*-
+# A part of NonVisual Desktop Access (NVDA)
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+# Copyright (C) 2019 NV Access Limited
+
+import subprocess
+from typing import Tuple, List
+
+_baseBranchPlaceholder = "<baseBranch>"
+_mergeBaseCommand = [
+	"git",
+	# merge-base is used to limit changes to only those that are new
+	# on HEAD.
+	"merge-base",
+	_baseBranchPlaceholder,  # this is the target branch used in a PR
+	"HEAD"
+]
+_mergeBasePlaceholder = "<mergeBase>"
+_diffCommand = [
+	"git", "diff",
+	# Only include changed lines (no context) in the diff. Otherwise
+	# developers may end up getting warnings for code adjacent to code they
+	# touched. This could result in very large change sets in order to get a
+	# clean build.
+	"-U0", _mergeBasePlaceholder
+	# We don't use triple dot syntax ('...') because it will not
+	# report changes in your working tree.
+]
+
+
+def substitutePlaceholder(
+		command: List[str], placeHolder: str, value: str
+) -> List[str]:
+	newList = command.copy()
+	placeHolderIndex = newList.index(placeHolder)
+	newList[placeHolderIndex] = value
+	return newList
+
+
+def getDiff(baseBranch: str) -> bytes:
+	mergeBaseCommand = substitutePlaceholder(_mergeBaseCommand, _baseBranchPlaceholder, baseBranch)
+	mergeBase: bytes = subprocess.check_output(mergeBaseCommand)
+	mergeBase: str = mergeBase.decode(encoding='ANSI').strip()

```

Comment:
Are you sure ANSI is the right encoding here? SHouldn't this be `locale.getpreferredencoding()` or locale.getlocale()[1]? Note that the former properly takes the Windows 10 beta unicode support into account.

Reply:
The output of this is a SHA and is to be consumed by an application. I think variations based on lang/locale would only be likely to cause problems.